src/map/status.cpp

```diff
@@ -9266,6 +9232,49 @@ int32 status_isimmune(struct block_list *bl)
 	return 0;
 }
 
+/**
+ * Returns whether object can be stopped by damage or not
+ * This does not only include the status change "Endure" but also all other endure effects
+ * @param bl: Object to check [PC|MOB|HOM|MER|ELEM]
+ * @param tick: Current tick
+ * @param visible: If true, function returns if the endure effect should be shown on the client
+ * @return Whether object can be stopped (false) or not (true)
+ */
+bool status_isendure(block_list& bl, t_tick tick, bool visible)
+{
+	// Officially the bonus "no_walk_delay" is actually just SC_ENDURE with unlimited duration
+	// Endure is forbidden on some maps, but we don't apply this on this bonus
+	// That's why we need to check it here
+	if (bl.m >= 0 && bl.type == BL_PC && !status_change_isDisabledOnMap(SC_ENDURE, map_getmapdata(bl.m))) {
```

Comment:
Is it okay to ignore status_change_isDisabledOnMap for other BL types?
In status_change_start we check it independent of BL type.

Reply:
special_state.no_walk_delay only applies to BL_PC

The restriction itself applies to all BL types of course, but it's not relevant here.